### PR TITLE
Added log10_ method to the PyTorch frontend.

### PR DESCRIPTION
**Pull request by he11owthere**
Closes https://github.com/unifyai/ivy/issues/17507

This pull request addresses the issue of missing inplace logarithm method in the PyTorch frontend. I have added the log10_ method to the torch_frontend.tensor module, which calculates the base-10 logarithm of the input tensor in-place. Additionally, I have added corresponding tests in test_tensor.py to ensure the correctness of the new method.

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by he11owthere** [comment]
I ensured that all tests are passing consistently on my local machine within the Docker environment.
For some reason these tests are failing.

File: .gitignore [hunk 1]
```diff
@@ -40,6 +40,4 @@ venv/
 ENV/
 env.bak/
 venv.bak/
-ivy_dev/
-*venv/
-venv*
\ No newline at end of file
+ivy_dev/
```
Comment (jkeane508) [inline .gitignore:43]: Can you please undo the changes here in this file :)

File: ivy/functional/frontends/torch/tensor.py [hunk 2]
```diff
@@ -1313,6 +1313,11 @@ def square(self):
     def log10(self):
         return torch_frontend.log10(self._ivy_array)
 
+    @with_unsupported_dtypes({"2.0.1 and below": ("float16",)}, "torch")
```
Comment (jkeane508) [inline ivy/functional/frontends/torch/tensor.py:1316]: Correct me if I'm wrong but if `log10` and `log10_` are essentially the same function could you just not make one an alias of the other?

**Review by jkeane508 — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/tensor.py (hunk 2 above)
Comment (he11owthere) [reply ivy/functional/frontends/torch/tensor.py:1316]: Thank you for the suggestion! However, "log10" and "log10_" have different behaviors - one is in-place, and the other creates a new tensor. Making them aliases might not be suitable in this case.

File: .gitignore (hunk 1 above)
Comment (he11owthere) [reply .gitignore]: Sure, I have reverted the changes in the gitignore file as you requested. The unnecessary line has been removed. Let me know if there's anything else I can assist with! :)

File: ivy/functional/frontends/torch/tensor.py (hunk 2 above)
Comment (jkeane508) [reply ivy/functional/frontends/torch/tensor.py:1316]: > Thank you for the suggestion! However, "log10" and "log10_" have different behaviors - one is in-place, and the other creates a new tensor. Making them aliases might not be suitable in this case.

Oh my bad! I must've missed that in the docs.

File: .gitignore [hunk 3]
```diff
@@ -42,4 +42,4 @@ env.bak/
 venv.bak/
 ivy_dev/
 *venv/
-venv*
\ No newline at end of file
+venv*
```
Comment (jkeane508) [inline .gitignore:45]: Not sure why this is throwing changes, but can you copy the raw text of the `.gitignore` on the Ivy Master Branch and paste it into your branches `.gitignore` and push that to this branch? This should remove differences.

**Review by jkeane508 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 3 above)
Comment (jkeane508) [reply .gitignore:45]: https://raw.githubusercontent.com/unifyai/ivy/master/.gitignore

Comment (he11owthere) [reply .gitignore]: Okay, I just copied the raw text of .gitignore from the Ivy master branch and pushed the changes. Now both .gitignore files are exactly the same. 😊

**Comment by jkeane508** [comment]
Just re-running tests because one of them failed :)

**Comment by he11owthere** [comment]
Thank you very much!!! I am truly grateful for the opportunity to contribute!!! 😄
